tests/sse/benchmark_sse1.cpp

```diff
@@ -0,0 +1,154 @@
+// Copyright 2014 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <xmmintrin.h>
+#include "benchmark_sse1.h"
+
+int dst_int = 0;
+
+int main()
+{
+	printf ("{ \"workload\": %u, \"results\": [\n", N);
+	assert(N%4 == 0); // Don't care about the tail for now.
+	float *src = alloc_float_buffer();//(float*)aligned_alloc(16, N*sizeof(float));
+	for(int i = 0; i < N; ++i)
+		src[i] = 1.0f + (float)rand() / (float)RAND_MAX;
+	float *src2 = alloc_float_buffer();//(float*)aligned_alloc(16, N*sizeof(float));
+	for(int i = 0; i < N; ++i)
+		src2[i] = 1.0f + (float)rand() / (float)RAND_MAX;
+	float *dst = alloc_float_buffer();//(float*)aligned_alloc(16, N*sizeof(float));
+
+	float scalarTime;
+	SETCHART("load");
+	START();
+		for(int i = 0; i < N; ++i)
+			dst[i] = src[i];
+	ENDSCALAR(checksum_dst(dst), "scalar");
+
+	LS_TEST("_mm_load_ps", _mm_load_ps, 0, _mm_store_ps, float*, 0, 4);
+	LS_TEST("_mm_load_ps1", _mm_load_ps1, 1, _mm_store_ps, float*, 0, 4);
+	LS_TEST("_mm_load_ss", _mm_load_ss, 1, _mm_store_ps, float*, 0, 4);
+	LS_TEST("_mm_load1_ps", _mm_load1_ps, 1, _mm_store_ps, float*, 0, 4);
+
+	__m128 tempReg = _mm_set_ps(1.f, 2.f, 3.f, 4.f);
+	LSH_TEST("_mm_loadh_pi", tempReg, _mm_loadh_pi, __m64*, 1, _mm_store_ps, float*, 0, 4);
+	LSH_TEST("_mm_loadl_pi", tempReg, _mm_loadh_pi, __m64*, 1, _mm_store_ps, float*, 0, 4);
+
+	LS_TEST("_mm_loadr_ps", _mm_loadr_ps, 0, _mm_store_ps, float*, 0, 4);
+	LS_TEST("_mm_loadu_ps", _mm_loadu_ps, 1, _mm_store_ps, float*, 0, 4);
+
+	SETCHART("set");
+	SS_TEST("_mm_set_ps", _mm_set_ps(src[i+2], src[i+1], src[i+5], src[i+0]));
+	SS_TEST("_mm_set_ps1", _mm_set_ps1(src[i]));
+	SS_TEST("_mm_set_ss", _mm_set_ss(src[i]));
+	SS_TEST("_mm_set1_ps", _mm_set1_ps(src[i]));
+	SS_TEST("_mm_setr_ps", _mm_set_ps(src[i+2], src[i+1], src[i+5], src[i+0]));
+	SS_TEST("_mm_setzero_ps", _mm_setzero_ps());
+
+	SETCHART("move");
+	SS_TEST("_mm_move_ss", _mm_move_ss(_mm_load_ps(src+i), _mm_load_ps(src2+i)));
+	SS_TEST("_mm_movehl_ps", _mm_movehl_ps(_mm_load_ps(src+i), _mm_load_ps(src2+i)));
+	SS_TEST("_mm_movelh_ps", _mm_movelh_ps(_mm_load_ps(src+i), _mm_load_ps(src2+i)));
+
+	SETCHART("movemask");
+	START(); for(int i = 0; i < N; i += 4) { int movemask = ((unsigned int)src[i] >> 31) | (((unsigned int)src[i+1] >> 30) & 2)  | (((unsigned int)src[i+2] >> 29) & 4)  | (((unsigned int)src[i+3] >> 28) & 8); dst_int += movemask; } ENDSCALAR(checksum_dst(&dst_int), "scalar movemask");
```

Comment:
Is there a reason this is all one line? I think it would be more readable if it were formatted normally. Same with other long lines with multiple statements below.

Reply:
The intent in these benchmark files have been to have each line implement its own instruction in full, to keep the large amount of code in this file low. I find that it's more navigatable that way overall, expanding these would make the file navigate more verbosely.

Because each line like this implements the matching SSE code in scalar, I find that it's not that important to have all of that "nice in sight", but more to hide it, so it feels more convenient to group them like this.